File: src/main/scala/edu/twitter/model_api/GenericModelBuilder.scala
package edu.twitter.model_api

/**
  * A generic interface for the object responsable for creating any model.
  */
trait GenericModelBuilder {
  /**
    * Run the recipe responsible for constructing the model.
    *
    * @return an instance of generic model.
    */
  def build(): GenericModel
}

File: src/main/java/edu/twitter/model/api/GenericModelBuilder.java
package edu.twitter.model.api;

/**
 * Generic interface for the object responsible for creating any model.
 */
public interface GenericModelBuilder {

    /**
     * Run the recipe responsible for constructing the model.
     *
     * @return an instance of generic model.
     */
    GenericModel build();
}
